featuretools/entityset/relationship.py

```diff
@@ -25,8 +25,8 @@ def __init__(self, entityset, parent_dataframe_id, parent_column_id,
 
         if (self.parent_dataframe.ww.index is not None and
                 self._parent_column_id != self.parent_dataframe.ww.index):
-            raise AttributeError(f"Parent column '{self.parent_column}' is not the index of "
-                                 f"dataframe {self.parent_dataframe}")
+            raise AttributeError(f"Parent column '{self.parent_column.name}' is not the index of "
+                                 f"dataframe {self.parent_dataframe.ww.name}")
```

Comment:
Not sure if it makes much difference, but can we use the relationship attributes here instead?
`self.parent_column.name` -> `self._parent_column_id`
`self.parent_dataframe.ww.name` -> `self._parent_dataframe_id`

Not sure if that is is guaranteed that the woodwork name will match the featuretools name for the dataframe? If not, we probably should try to stick with the name that is used by Featuretools.

Reply:
I think you're probably right that going through the relationship attributes makes sense here even if we were 100% confident that the names are the same. 

Regarding whether the Woodwork name will match the featuretools name, `EntitySet.add_name` currently requires a `dataframe_id` and I've made so that even if a user passes a Woodwork dataframe in whose name doesn't match, we set `dataframe.ww.name = dataframe_id` so that `es[df_id].ww.name` should always match `df_id` (I think)